packages/core/src/Popout/Popout.tsx

```diff
@@ -0,0 +1,134 @@
+  position: relative;
+  width: fit-content;
+`
+
+const OnClickHandler = styled(Box)`
+  position: absolute;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  cursor: pointer;
+`
+
+const Shadow = styled(Box)`
+  position: fixed;
+  top: 0;
+  left: 0;
+  cursor: pointer;
+  width: 100vw;
+  height: 100vh;
+  z-index: 1;
+  transition: opacity 250ms ease-in-out;
+`
+
+const PopoutModal = styled(Box)`
+  position: absolute;
+  z-index: 2;
+  background-color: white;
+  transition: padding 200ms ease-in-out, margin 200ms ease-in-out, border-radius 200ms ease-in-out;
+`
+
+export const Popout = (props: IPopoutProps) => {
+  const { trigger, content, onOpen, onClose, triggerRef, closeOnTriggerRefClick } = props
+  const [isOpen, setIsOpen] = useState(false)
+  const [height, setHeight] = useState(0)
+  const [padding, setPadding] = useState(0)
+  const [opacity, setOpacity] = useState(0)
+  const baseInputRef = useRef<HTMLDivElement>(null)
+  const modalRef = useRef<HTMLDivElement>(null)
+
+  const handleOpen = useCallback(() => {
+    setIsOpen(true)
+    setTimeout(() => {
+      setPadding(16)
+      setOpacity(0.5)
+      /* istanbul ignore next */
+      triggerRef?.current?.focus()
+      /* istanbul ignore next */
+      onOpen?.()
+    })
+  }, [triggerRef, onOpen])
+
+  const handleClose = useCallback(() => {
+    setIsOpen(false)
+    setTimeout(() => {
+      setPadding(0)
+      setOpacity(0)
+      /* istanbul ignore next */
+      onClose?.()
+    })
+  }, [onClose])
+
+  useLayoutEffect(() => {
+    setHeight(baseInputRef.current.clientHeight)
+  }, [trigger])
+
+  useEffect(() => {
+    if (isOpen && closeOnTriggerRefClick) triggerRef?.current?.addEventListener('click', handleClose)
+    return () => triggerRef?.current?.removeEventListener('click', handleClose)
+  }, [isOpen, closeOnTriggerRefClick, triggerRef, handleClose])
+
+  useEffect(() => {
+    const escapeListener = (e: KeyboardEvent) => {
+      if (isOpen && e.key === 'Escape') handleClose()
+    }
+    const focusListener = (e: FocusEvent) => {
+      if (
+        isOpen &&
+        e.target instanceof HTMLElement &&
+        e.target !== baseInputRef.current &&
+        !modalRef.current.contains(e.target)
+      )
+        handleClose()
+    }
+    document.addEventListener('focusin', focusListener)
+    document.addEventListener('keydown', escapeListener)
+    return () => {
+      document.removeEventListener('focusin', focusListener)
+      document.removeEventListener('keydown', escapeListener)
+    }
+  }, [isOpen, handleClose])
+
+  const placeHolderDiv = <Box style={{ height }} />
+
+  return (
+    <Box>
+      <TriggerWrapper tabIndex={isOpen ? -1 : 0} onFocus={handleOpen} ref={baseInputRef}>
+        {isOpen ? placeHolderDiv : trigger}
+        <OnClickHandler onClick={handleOpen} />
+      </TriggerWrapper>
+      {isOpen && (
+        <>
+          <Shadow bg='background.darkest' style={{ opacity }} onClick={handleClose} />
```

Comment:
I love adding consistency throughout the Design-System! I used the `ShadowOverlay` that is exported alongside the `ShadowEffect` component, since its use case is exactly the same as what I needed here. Works like a charm, thanks for pointing it out!